frontend/class-opengraph-image.php

```diff
@@ -52,11 +52,22 @@ class WPSEO_OpenGraph_Image {
 	/**
 	 * Constructor.
 	 *
-	 * @param WPSEO_OpenGraph $wpseo_opengraph The OpenGraph object.
+	 * @param null|string     $image           Optional. The Image to use.
+	 * @param WPSEO_OpenGraph $wpseo_opengraph Optional. The OpenGraph object.
 	 */
-	public function __construct( WPSEO_OpenGraph $wpseo_opengraph ) {
+	public function __construct( $image = null, WPSEO_OpenGraph $wpseo_opengraph = null ) {
```

Comment:
I think it would be better to keep the current interface, but deprecate the argument type string instead. Or else you'd have to keep passing null as a first argument _just_ for BC reasons.

How's this?

``` php
/** 
 * Constructor.
 *
 *  @param WPSEO_OpenGraph|null|string $wpseo_opengraph Optional. The OpenGraph object.
 */
public function __construct( $wpseo_opengraph = null ) {
    if ( $wpseo_opengraph === null || ! $wpseo_opengraph instanceof WPSEO_OpenGraph ) {
			$wpseo_opengraph = new WPSEO_OpenGraph();
		}
    $this->opengraph = $wpseo_opengraph

    if ( is_string( $wpseo_opengraph ) ) {
        _deprecated_argument( __METHOD__, 'WPSEO 7.4', 'Pass an instance of WPSEO_OpenGraph instead of the path to an image' );
        if( !empty( $wpseo_opengraph) ) {
    		$this->add_image_by_url( $image );
        }
	}

    $this->set_images();
}
```

Reply:
This situation is an exception, the intended use is via `WPSEO_OpenGraph::image` to add an image.
In that method the `image` and `this` of the `WPSEO_OpenGraph` instance.

We want to avoid creating more instances of the `WPSEO_OpenGraph`, because all the hooks are being registered in the constructor.